components/Sequence/models.rst

```diff
@@ -0,0 +1,101 @@
+Models
+======
+
+.. _component_sequence_model_sequence:
+
+Sequence
+--------
+
+**Sequence** is a generated set of numbers and/or letters stored on a model. They may be useful for example as order identifiers.
+
++------------+------------------------------+
+| Property   | Description                  |
++============+==============================+
+| id         | Unique id of the sequence    |
++------------+------------------------------+
+| index      | The sequence's index         |
++------------+------------------------------+
+| type       | The sequence's type          |
++------------+------------------------------+
+
+.. note::
+   This model implements the :ref:`component_sequence_model_sequence-interface`. |br|
+   For more detailed information go to `Sylius API Sequence`_.
+
+.. _Sylius API Sequence: http://api.sylius.org/Sylius/Component/Sequence/Model/Sequence.html
+
+Generators
+----------
+
+.. _component_sequence_number_abstract-generator:
+
+AbstractGenerator
+~~~~~~~~~~~~~~~~~
+
+A custom generator model should extend this class in order to generate sequences
+on a given object that implements the :ref:`component_sequence_model_sequence-subject-interface`.
+
+.. note::
+   This model implements the :ref:`component_sequence_number_generator-interface`. |br|
+   For more detailed information go to `Sylius API AbstractGenerator`_.
+
+.. _Sylius API AbstractGenerator: http://api.sylius.org/Sylius/Component/Sequence/Number/AbstractGenerator.html
+
+.. _component_sequence_number_hash-generator:
+
+HashGenerator
+~~~~~~~~~~~~~
+
+This class generates hash numbers similar to the Amazon order identifiers (e.g. 105-3958356-3707476)
+and also random hashed segments of a given length.
+
++--------------------+------------------------------+
+| Property           | Description                  |
++====================+==============================+
+| subjectRepository  | The hash subject repository  |
++--------------------+------------------------------+
+
+.. note::
+   This model implements the :ref:`component_sequence_number_generator-interface`
+   and extends the :ref:`component_sequence_number_abstract-generator`.|br|
+   For more detailed information go to `Sylius API HashGenerator`_.
+
+.. _Sylius API HashGenerator: http://api.sylius.org/Sylius/Component/Sequence/Number/HashGenerator.html
+
+.. _component_sequence_number_sequential-generator:
+
+SequentialGenerator
+~~~~~~~~~~~~~~~~~~~
+
+This class is a default order number generator.
+
++--------------------+------------------------------+
+| Property           | Description                  |
++====================+==============================+
+| numberLength       | Order number max length      |
++--------------------+------------------------------+
+| startNumber        | The sequence start number    |
```

Comment:
I'd suggest showing here code example here, with these options passed to constructor. Listing properties is not that helpful, they should be in api docs anyway.